src/services/authService.js

```diff
@@ -30,11 +32,73 @@ export default class AuthService {
     return { token, ...Helper.omitFields(user, ['password']) };
   }
 
+  /**
+ *
+ * @param {string} email email address
+ * @returns {string} token
+ */
+  static async forgotPassword(email) {
+    const result = await User.findOne({ where: { email } });
+
+    if (!result) throw new Error('Email not found');
+
+    const { dataValues: { first_name: name, } } = result;
+    const token = await jwt.sign({
+      email
+    }, jwtSecret, { expiresIn: '1h' });
+    const emailDetails = {
+      receiver: email,
+      sender: process.env.SENDER,
+      templateName: 'reset_password',
+      name,
+      url: `provide_newpass/reset_password/${token}`
+    };
+
+    await sendmail(emailDetails);
+
+    await User.update({
+      reset_token: token
+    }, {
+      where: {
+        email
+      }
+    });
+
+    return { token };
+  }
+
+  /**
+   *
+   * @param {string} token request token
+   * @param {string} password new password
+   * @returns {string} message
+   * */
+  static async resetPassword(token, password) {
+    const { email } = await jwt.verify(token, jwtSecret);
+    const { dataValues: { reset_token: tokenSaved } } = await User.findOne({ where: { email } });
+
+    if (token !== tokenSaved) throw new Error('Invalid token');
+    if (password === undefined || password.trim().length < 8) throw new Error('Provide valid password');
```

Comment:
You should use our validation schema to check password exist and the length of the password. @cleave3 has implemented this on his task, reach out to him to guide you